### ISSUE TITLE
Need to set default apps for opeening from ranger

### ISSUE THREAD

**Issue by Harighs** [ranger/ranger#2448]
Hello i need to set my default opening software from ranger for these files:

.py -> pycharm
.txt -> ubuntu default txt editor
.pdf -. default document viewer 

can you help me how to do it. should i want to change something in rifle.conf file?

**Comment by toonn** [comment]
Yep, just change the appropriate rifle rules, the file has relevant information about how rules work at the top. Rifle falls back to xdg-open if there are no matching rules for a file, so you could also configure that and remove the offending rules. For the text editor you'd normally set `VISUAL` as an env var in your shell. Do note that ranger expects this editor *not* to fork so GUIs are out of the question. If you want to use a GUI editor anyway, make sure the rule with `label editor` is a terminal editor that doesn't fork, vim, emacs, nano.

Questions like this are much better suited to the r/ranger subreddit, not the issue tracker.

Have a nice day!
